h2o-core/src/main/java/hex/Model.java

```diff
@@ -1614,7 +1622,7 @@ protected void toJavaPredictBody(SBPrintStream body,
                                    CodeGeneratorPipeline classCtx,
                                    CodeGeneratorPipeline fileCtx,
                                    boolean verboseCode) {
-    throw new IllegalArgumentException("This model type does not support conversion to Java");
+    throw new UnsupportedConversionToJavaException("This model type does not support conversion to Java");
```

Comment:
I don't think this should really be an IllegalArgumentException. Better exception seems to be UnsupportedOperationException. If we need a custom exception here, consider inheriting from H2OAbstractRuntimeException.

Reply:
throwing `UnsupportedOperationException`